software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java

```diff
@@ -317,6 +331,8 @@ protected void computePortsAndUrls() {
 
         sensors().set(PROTOCOL, inferProtocol());
         sensors().set(MAIN_URI, URI.create(inferUrl()));
+        sensors().set(MAIN_URI_MAPPED_SUBNET, URI.create("http://" + sensors().get(SUBNET_ADDRESS) + ":" + sensors().get(PROXY_HTTP_PORT) + "/"));
+        sensors().set(MAIN_URI_MAPPED_PUBLIC, URI.create(inferUrlPublic()));
```

Comment:
@neykov Will the `On{Public,Private}NetworkEnricher` returns the right value? The changes I made was an attempt to follow @ahgittin's proposal